pkg/cluster/cluster.go

```diff
@@ -118,8 +118,12 @@ func (c *Cluster) setup() error {
 		return fmt.Errorf("invalid cluster spec: %v", err)
 	}
 
+	var phase spec.ClusterPhase
+	if c.cluster.Status != nil {
+		phase = c.cluster.Status.Phase
+	}
 	var shouldCreateCluster bool
-	switch c.status.Phase {
+	switch phase {
 	case "":
```

Comment:
i dislike all undefined string or number. but this is a minor thing. i can clean it up later.